### PR TITLE
Fix all torch pipeline failures except one

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Fix most torch pipeline failures. See comments in the changes

### CODE REVIEW

File: tests/pipelines/test_pipelines_automatic_speech_recognition.py [hunk 1]
```diff
@@ -1216,7 +1216,7 @@ def test_with_local_lm_fast(self):
...
+        EXPECTED_RESULT = " Folks, if you watch the show, you know, I spent a lot of time right over there. Patiently and astutely scrutinizing the boxwood and mahogany chest set of the day's biggest stories developing the central headline pawns, definitely maneuvering an oso topical night to F6, fainting a classic Sicilian, nade door variation on the news, all the while seeing eight moves deep and patiently marshalling the latest press releases into a fisher's shows in Lip Nitsky attack that culminates in the elegant lethal slow-played, all-passant checkmate that is my nightly monologue. But sometimes, sometimes, folks, I. CHEERING AND APPLAUSE Sometimes I startle away, cubside down in the monkey bars of a condemned playground on a super fun site. Get all hept up on goofballs. Rummage that were discarded tag bag of defective toys. Yank out a fist bowl of disembodied doll limbs, toss them on Saturday, Rusty Cargo, container down by the Wharf, and challenge toothless drifters to the godless bughouse lets of tournament that is my segment. MUSIC Meanwhile!"
```
Comment (ydshieh) [inline tests/pipelines/test_pipelines_automatic_speech_recognition.py:1219]: this is sensible. Similar change is done in  #30152

File: tests/pipelines/test_pipelines_document_question_answering.py [hunk 2]
```diff
@@ -103,7 +103,7 @@ def run_pipeline_test(self, dqa_pipeline, examples):
     @require_detectron2
     @require_pytesseract
     def test_small_model_pt(self):
-        dqa_pipeline = pipeline("document-question-answering", model="hf-internal-testing/tiny-random-layoutlmv2")
```
Comment (ydshieh) [inline tests/pipelines/test_pipelines_document_question_answering.py:106]: this checkpoint doesn't contain all weights, and random init. doesn't reproduce in each run.
(`LayoutLMv2PreTrainedModel._init_weights` is not written to take into account `LayoutLMv2VisualBackbone` added later)

File: tests/pipelines/test_pipelines_document_question_answering.py [hunk 3]
```diff
@@ -103,7 +103,7 @@ def run_pipeline_test(self, dqa_pipeline, examples):
...
+        dqa_pipeline = pipeline("document-question-answering", model="hf-internal-testing/tiny-random-layoutlmv2-for-dqa-test")
```
Comment (ydshieh) [inline tests/pipelines/test_pipelines_document_question_answering.py:106]: So I just load the checkpoint, save it and upload to another repo then use it

File: tests/pipelines/test_pipelines_zero_shot_audio_classification.py [hunk 4]
```diff
@@ -32,7 +32,7 @@ def test_small_model_pt(self):
         audio_classifier = pipeline(
             task="zero-shot-audio-classification", model="hf-internal-testing/tiny-clap-htsat-unfused"
         )
-        dataset = load_dataset("ashraq/esc50")
+        dataset = load_dataset("hf-internal-testing/ashraq-esc50-1-dog-example")
```
Comment (ydshieh) [inline tests/pipelines/test_pipelines_zero_shot_audio_classification.py:35]: this is done in another PR.  It is changed here again just for me to proceed fast while waiting the review of another PR.

I will resolve the conflict before merge.

**Review by amyeroberts — APPROVED** [review]
Thanks for handling all of these!

**Pull request merged**